python/cudf/cudf/core/dataframe.py

```diff
@@ -4071,7 +4071,7 @@ def scatter_by_map(self, map_index, map_size=None, keep_index=True):
 
         # scatter_to_frames wants a list of columns
         tables = libcudf.copying.scatter_to_frames(
-            self._columns, map_index._column, index
+            self.iloc[:]._columns, map_index._column, index
```

Comment:
Would really like to eliminate this `iloc` here as it shouldn't be necessary, did the copy of the `source` list not help?